cmd/antrea-agent/agent.go

```diff
@@ -999,6 +992,14 @@ func run(o *Options) error {
 		go nodeLatencyMonitor.Run(stopCh)
 	}
 
+	if egressController != nil {
+		go egressController.Run(stopCh)
+	}
+
+	if externalIPController != nil {
+		go externalIPController.Run(stopCh)
+	}
+
```

Comment:
My concern with changing the order is that it is a bit arbitrary, we can introduce new unexpected issues, and it limits what we can do in the future. For example, we could in the future want to introduce a dependency of `flowRestoreCompleteWait` on the realization of Egress policies. It would make sense: delay the removal of `flow-restore-wait` until Egress policy flows have been installed, in order to provide a more consistent datapath on (re)start. See https://github.com/antrea-io/antrea/pull/6342 for more context.

However, we know that there is already a dependency of SecondaryNetwork initialization on `flowRestoreCompleteWait`. This dependency is important and AFAIK cannot be broken. So with the change described above, we would end up with a circular dependency:
`EgressController` before `flowRestoreCompleteWait` before `SecondaryNetwork initialization` before `EgressController`.

I would rather avoid "introducing" this new dependency (or rather enforcing this new dependency).

cc @tnqn for his opinion

Reply:
I think this is a valid concern. we can check/watch for interface changes in the responders to avoid the hard dependencies. Waiting for Quan's insights.